src/fuzzy_match.cc

```diff
@@ -579,12 +582,13 @@ namespace fuzzy
       if (p_length <= agendaItem.coverage + nGramMatches.max_differences_with_pattern)
       {
         Costs costs;
-        costs.diff_word = 100. / std::max(suffix_wids.size(), p_length);
+        costs.diff_word = 100. / std::max(s_length, p_length);
 
         /* let us check the candidates */
         const auto suffix_realtok = _suffixArrayIndex->real_tokens(s_id);
         float cost = _edit_distance(suffix_wids, suffix_realtok,
-                                    pattern_wids, pattern_realtok,
+                                    s_length,
```

Comment:
I suggest putting `suffix_wids, suffix_realtok, s_length` on the same line, and `pattern_wids.data(), pattern_realtok, p_length` on another to better group related parameters.